doc/reflection-milestone2.md

```diff
@@ -1,43 +1,46 @@
-reflection-milestone2
+Reflection - Milestone2
 ================
 Cheuk Ho, Hazel Jiang, Anita Li, Ivy Zhang
 22/01/2021
 
-In this milestone, we implemented the 4 key factors (`Monthly Income`,
+In this milestone, we implemented the plots for the 4 key factors (`Monthly Income`,
 `Work Life Balance`, `Business Travel` and `Environment Satisfaction`)
-that we believe will contribute to employee attrition based on IBM
-employee satisfaction survey data. We also implemented 2 dropdown
-filters (`Department` and `Gender`) and 1 slide bar (`Age`) for the
-management team to play with in order to see whether the attrition rate
-vary between features. By filtering the data, our plots show the
-proportion of attrition among the 4 key factors, which could help the
-management team to make better decisions.
+that contributes to employee attrition based on IBM
+employee satisfaction survey data. We also implemented a total of 
+3 interactive widgets to our app, 2 dropdown filters (`Department` and `Gender`) 
+and 1 slide bar (`Age`) for the management team to play with in order to see 
+whether the attrition rate vary between features. By filtering the data, our plots 
+show the attrition among the 4 key factors among different employee segment, which could help the
+management team to develop their action plan for next step.
 
-There are two things we have in mind but not yet implemented
-successfully. We have four widgets in our proposal but we are only able
-to implement three of them. We left out `Job Role` feature because this
-feature has interaction with `Department` feature. The selected job role
-has to be included in the selected department for the dashboard to have
-data to plot, otherwise we could only get empty plots. Therefore, in
-this milestone, we dropped `Job Role` and only kept `Department` in the
+We have four widgets in our proposal but we decided to only 
+implement three of them. We left out `Job Role` feature because this
+feature is depending on `Department` feature. The selected job role
+needs to be belong to the selected department for the dashboard to have
```

Comment:
Suggestion to consider: We left out `Job Role` to prevent the dashboard from showing blanks, which would negatively impact the user experience. This may occur for certain roles especially when limited data associated to them.  Since these roles can be regrouped into a department oriented variable, we consider the `Job Role` feature highly depends on the `Department` feature. As a result, we decided to only keep `Department` in our widget list for this milestone.